docs/errors.md

```diff
@@ -0,0 +1,265 @@
+
+"Resolution" refers to converting a specifier into a URL. This happens in any of the following scenarios:
+
+- a module is loaded directly - `System.import('specifier')`
+- a module is loaded as a dependency - `System.register(['specifier'], ...)` or `define(['specifier'], ...)`
+- a module is resolved manually - `System.resolve('specifier')`
+
+To fix this error, you may either use [import maps](/docs/import-maps.md) or [hook System.resolve](/docs/hooks.md#resolveid-parenturl---string):
+
+```html
+<script src="/system.js"></script>
+<script>
+  (function () {
+    const originalResolve = System.constructor.prototype.resolve;
+    System.constructor.prototype.resolve = function (id) {
+      try {
+        return originalResolve.apply(this, arguments);
+      } catch (err) {
+        return otherResolution(id);
+      }
+    };
+
+    function otherResolution (id) {
+      return '/some-url.js';
+    }
+  })()
+</script>
+```
+
+## 4
+
+### Unable to load module
+
+SystemJS Error #4 occurs when a module could not be downloaded and/or executed.
+
+SystemJS loads modules by appending `<script>` elements to the DOM and waiting for `load` and `error` events. When error #4 is thrown, the script's `error` event fired.
+
+Here are common reasons why a module could not be downloaded:
+
+- **The url for the module is incorrect**. You can check this by opening up the module's URL in its own browser tab and verifying that a javascript file is shown. The module URL can be found in the browser's devtool console and network tab, or by calling `System.resolve('name')` in the browser console to view the full URL. Be sure to check port number! Also, verify that the URL returns javascript instead of HTML.
+- **The url is correct, but the web server that hosts it isn't running**. This often occurs when you're attempting to download a module from localhost, but haven't started up the web server (ie webpack-dev-server or `npm start`). Be sure to check the port number!
+- **The url is correct and the web server is running, but CORS is not enabled**. [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) is a security mechanism that browsers use to protect users. The server you are downloading the module from may need to enable cors by sending an [`Access-Control-Allow-Origin` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin).
+- **The url is correct, web server running, CORS enabled, but host check fails**. This one is usually specific to webpack-dev-server, which has a [host check](https://webpack.js.org/configuration/dev-server/#devserverdisablehostcheck). You should disable that host check to be able to load modules cross origin.
+- **The javascript file was successfully downloaded, but failed during initial execution**. A syntax error, errant function call, or any other javascript error that occurs during initial execution could cause the module to fail to load.
```

Comment:
Can we clarify "initial **top-level** execution". Maybe worth noting the original error can be found in the browser console here?